### PR TITLE
TokenRepository improvements

### PR DESCRIPTION
**Pull request by terranmoccasin**
https://app.asana.com/0/1204730191677489/1205117637172628/f

- Fetch methods return an additional flag indicating whether the token exists in the repository instance. This augments the TokenFetcher fetch methods with results derived from the context of the repository.
- Removed exclude functionality. Since fetch results have additional metadata that callers use for filtering, exclusion is now the responsibility of the caller to manage with their own tags - e.g. add "blocked" tag and filter out those results.
- Added ability to tags to mints without adding them to the repository.

### CODE REVIEW

File: packages/token-sdk/src/repository.ts [hunk 1]
```diff
@@ -149,18 +150,9 @@ export class TokenRepository {
     fetcher: TokenFetcher,
     mints: Address[],
     refresh = false
-  ): Promise<TokenWithTags[]> {
```
Comment (yugure-orca) [inline packages/token-sdk/src/repository.ts:152]: nit: I think we can remove "Excluded mints are not returned." from the comment.

File: packages/token-sdk/src/repository.ts [hunk 2]
```diff
@@ -173,7 +165,7 @@ export class TokenRepository {
    * @returns Token metadata and tags for all mints with the given tag. Excluded mints are not
    * returned.
```
Comment (yugure-orca) [inline packages/token-sdk/src/repository.ts:166]: I think the excluded feature has been removed.
Wouldn't a filter based on ``exists`` be necessary to implement "If a mint is not in the repository, it is not returned"?

File: packages/token-sdk/src/repository.ts [hunk 3]
```diff
@@ -192,8 +184,8 @@ export class TokenRepository {
     if (tag === undefined) {
       return this.mintMap.has(mintString);
     }
-    const tagSet = this.mintMap.get(mintString);
-    return tagSet?.has(tag) ?? false;
+    const entry = this.mintMap.get(mintString);
+    return entry?.tags.has(tag) ?? false;
```
Comment (yugure-orca) [inline packages/token-sdk/src/repository.ts:188]: nit: If ``has = true``, the mint included in fetchAll seems intuitive. How about checking ``exists``.

**Review by yugure-orca — APPROVED** [review]
Thanks for the improvements!
I leave some comments, but please feel free to merge after small updates if required.

File: packages/token-sdk/src/repository.ts (hunk 3 above)
Comment (terranmoccasin) [reply packages/token-sdk/src/repository.ts]: This is a great catch - thank you. I've corrected and added it to a test.

**Pull request merged**